src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,120 @@
 package com.es.phoneshop.model.product;
 
-import java.util.List;
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
+
+    private static long MAX_ID;
+    private static final List<Product> products = new ArrayList<>();
```

Comment:
There is not really any point in making this field static (and hence initializing it from a static context)